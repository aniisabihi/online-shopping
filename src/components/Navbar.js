import React from "react";
import { Link } from "react-router-dom";

const Navbar = () => {
  return (
    <nav className="nav-wrapper">
      <div className="container">
        <Link to="/" className="brand-logo" title="Home">
          Online Shopping
        </Link>

        <Link to="/cart" className="right" title="View cart">
          <i className="material-icons">shopping_cart</i>
        </Link>
      </div>
    </nav>
  );
};

export default Navbar;
