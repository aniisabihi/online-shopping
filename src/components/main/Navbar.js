import React from "react";
import { Link } from "react-router-dom";

const Navbar = () => {
  return (
    <nav className="nav-wrapper">
      <div className="container">
        <Link
          to="/"
          id="home-button"
          className="brand-logo waves-effect waves-light"
          title="Home"
        >
          Online Shopping
        </Link>

        <Link
          to="/cart"
          id="cart-button"
          className="right waves-effect waves-light"
          title="View cart"
        >
          <i className="material-icons">shopping_cart</i>
        </Link>
      </div>
    </nav>
  );
};

export default Navbar;
