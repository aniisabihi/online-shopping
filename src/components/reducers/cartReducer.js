import items from './../../products.json';

const initState = {
    items: [items],
    addedItems: [],
    total: 0
}

const cartReducer= (state = initState, action) => {

    return state;
}

export default cartReducer;
