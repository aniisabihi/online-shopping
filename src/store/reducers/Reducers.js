import { combineReducers } from "redux";
import cartReducer from "./slices/cartSlice";

const data = require("./../../data.json");

export const reducers = combineReducers({
  products: data,
  cart: cartReducer,
});

export default data;
