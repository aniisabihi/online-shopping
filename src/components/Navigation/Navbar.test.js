import React from "react";
import { shallow } from "enzyme";
import Navbar from "./Navbar";
import { configureStore } from "@reduxjs/toolkit";
import { reducers } from "../../store/reducers/Reducers";
import { Provider } from "react-redux";

const NavbarComponent = <Navbar />;
const store = configureStore({ reducer: reducers });

let navbar = shallow(<Provider store={store}>{NavbarComponent}</Provider>);

/* RENDER COMPONENT */
describe("Render Navigation", () => {
  it("should render the Navigation Component accordingly", () => {
    expect(navbar).toMatchSnapshot();
  });
});
