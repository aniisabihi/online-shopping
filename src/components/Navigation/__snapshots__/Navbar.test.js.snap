// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Navigation should render the Navigation Component accordingly 1`] = `ShallowWrapper {}`;
